src/foam/core/pii/PIIReportRequestView.js

```diff
@@ -36,36 +78,93 @@ foam.CLASS({
   `,
 
   messages: [
-    { name: 'TITLE', message: 'PII Report Request' },
-    { name: 'MSG', message: 'A PII Report has been emailed to ${email}.', template: true }
+    { name: 'TITLE', message: 'Personal Identifiable Information Report' },
+    { name: 'HELP_TEXT_1', message: 'In compliance with General Data Protection Regulations (GDPR), this service will generate a report containing Personal Identifiable Information (PII) held by this system.' },
+    { name: 'HELP_TEXT_2', message: 'The report will be emailed to ${email}.', template: true },
+    { name: 'HELP_TEXT_3', message: 'Only one report will be generated in a ${requestTtl} hour period.', template: true },
+    { name: 'MSG_CONFIRMATION', message: 'A PII Report has been emailed to ${email}.', template: true },
+    { name: 'MSG_ALREADY', message: 'A PII Report has been requested recently, you can request another tomorrow.' }
+  ],
+
+  properties: [
+    {
+      documentation: 'Allow one request per time window - in hours',
+      name: 'requestTtl',
+      class: 'Long',
+      units: 'h',
+      value: 24
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.pii.PIIReportRequestView.GenerateRequest',
+      name: 'data',
+      factory: function() {
+        return this.GenerateRequest.create({parent: this});
+      }
+    },
+    {
+      name: 'sent',
```

Comment:
@sarthak-marwaha require help with this property.  When the generate button selected and the ticket is generated and closed, I want the button to be removed and MSG_CONFIRMATION displayed. 
The dynamic function clears the button, the msg is not displayed.